pkg/clusterapi/aws/utils.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"fmt"
+	"sort"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+
+	cov1 "github.com/openshift/cluster-operator/pkg/apis/clusteroperator/v1alpha1"
+)
+
+// SortInstances sorts the given slice of instances based on their launch time. The first item in the
+// slice after sorting will be the most recently launched, and can be considered the definitive instance
+// for the machine, a caller may wish to terminate all others. This function should only be called with
+// running instances, not those which are stopped or terminated.
+func SortInstances(instances []*ec2.Instance) {
```

Comment:
Is there any reason to fully sort the instances? All we want is the newest instance, right? Can we simply return the newest instance or the index in the slice for the newest instance?

Reply:
This was based on a request from Cesar in previous PR, after implementing it, it did seem a little simpler in the code when it comes time to figure out what instance to use, and obtain a slice to delete. This does however assume we're not getting into scenarios where we have thousands of instances existing for a single machine.

Reply:
Whic are we trying to make simpler, the caller code or the function code? If it is the caller code, then can we have this function swap the newest instance to the front of the slice?